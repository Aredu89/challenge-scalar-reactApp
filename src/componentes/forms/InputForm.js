import React from 'react';
import TextField from '@material-ui/core/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';
import './InputForm.css';

const InputForm = props =>{
  //Type puede ser: "password"
  const {
    label,
    type,
    onChange,
    value,
    error,
    errorText,
    inputProps,
    simbolo,
    multiline,
    rows
  } = props

  const auxInputProps = {
    ...inputProps
  }

  if(simbolo){
    auxInputProps.startAdornment = <InputAdornment position="start">{simbolo}</InputAdornment>
  }

  return(
    <div className="input-form w-100">
      <TextField
        id={label ? label.toString() : 'input-form'} 
        label={label}
        variant="outlined"
        type={type ? type : ""}
        onChange={event=>onChange(event)}
        value={value}
        error={error ? error : false}
        helperText={errorText && error ? errorText : ""}
        InputProps={auxInputProps}
        inputProps={auxInputProps}
        multiline={multiline ? true : false}
        rows={rows ? rows : 4}
        />
    </div>
  )
}

export default InputForm