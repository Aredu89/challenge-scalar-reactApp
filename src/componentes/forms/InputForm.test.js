import React from 'react';
import { render, cleanup } from '@testing-library/react';
import InputForm from './InputForm';

afterEach(cleanup)

describe('InputForm tests', ()=> {
  it('InputForm Snapshot', () => {
    const { asFragment } = render(<InputForm />);
    expect(asFragment(<InputForm />)).toMatchSnapshot()
  });
})