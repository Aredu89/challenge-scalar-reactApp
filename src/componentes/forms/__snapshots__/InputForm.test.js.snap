// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm tests InputForm Snapshot 1`] = `
<DocumentFragment>
  <div
    class="input-form w-100"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          id="input-form"
          rows="4"
          type=""
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-2"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
